java-frontend/src/main/java/org/sonar/java/se/checks/FlowComputation.java

```diff
@@ -125,19 +125,28 @@ private Symbol flowFromLearnedSymbols(ExplodedGraph.Node currentNode, @Nullable
     if (trackSymbol == null || parent == null) {
       return null;
     }
-    Optional<Symbol> learnedSymbol = currentNode.getLearnedSymbols().stream()
-      .map(ExplodedGraph.Node.LearnedValue::getSymbol)
-      .filter(symbol -> symbol.equals(trackSymbol))
+    Optional<ExplodedGraph.Node.LearnedValue> learnedValue = currentNode.getLearnedSymbols().stream()
+      .filter(lv -> lv.getSymbol().equals(trackSymbol))
       .findFirst();
-    if (learnedSymbol.isPresent()) {
-      flow.add(location(parent));
+    if (learnedValue.isPresent()) {
+      ExplodedGraph.Node.LearnedValue lv = learnedValue.get();
+      Constraint constraintForSV = parent.programState.getConstraint(lv.getSv());
+      if (constraintForSV == null) {
```

Comment:
I would go for a simpler code with sthg like :
```
 String message = "...";
if(constraintForSv != null) {
  message = ... 
}
flow.add(.... )  
```
That would allow to get rid of one of the static method.